### PR TITLE
Add HalfNormal prior distribution for non-negative variables.

### PR DESCRIPTION
**Pull request by ZitongZhou**
Halfnormal distribution could be useful for variables that are constrained to be non-negative.

### CODE REVIEW

File: gpytorch/priors/torch_priors.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #!/usr/bin/env python3
 
 import torch
-from torch.distributions import Gamma, HalfCauchy, LogNormal, MultivariateNormal, Normal, Uniform
+from torch.distributions import Gamma, HalfCauchy, LogNormal, MultivariateNormal, Normal, Uniform, HalfNormal
```
Comment (feynmanliang) [inline gpytorch/priors/torch_priors.py:4]: alphabetize

File: gpytorch/priors/torch_priors.py [hunk 2]
```diff
@@ -30,6 +30,28 @@ def expand(self, batch_shape):
         return NormalPrior(self.loc.expand(batch_shape), self.scale.expand(batch_shape))
 
 
+class HalfNormalPrior(Prior, HalfNormal):
+    """
+    Half-Normal prior.
+    
+    pdf(x) = 2 * (2 * pi * sigma^2)^-0.5 * exp(-x^2 / (2 * sigma^2)) for x >= 0; 0 for x < 0
+
+    where sigma^2 is the variance.
```
Comment (feynmanliang) [inline gpytorch/priors/torch_priors.py:39]: nit: can we use `scale` or `sigma` consistently between documentation and method arguments?

**Comment by feynmanliang** [comment]
Awesome, thanks!

File: gpytorch/priors/torch_priors.py [hunk 3]
```diff
@@ -30,6 +30,28 @@ def expand(self, batch_shape):
...
+    def __init__(self, scale, validate_args=None, transform=None):
+        TModule.__init__(self)
+        HalfNormal.__init__(self, scale=scale, validate_args=validate_args)
+        self._transform = transform
+
+    def expand(self, batch_shape):
+        return HalfNormal(self.scale.expand(batch_shape))
+
+    def __call__(self, *args, **kwargs):
+        return super(HalfNormal, self).__call__(*args, **kwargs)
```
Comment (Balandat) [inline gpytorch/priors/torch_priors.py:52]: why is this necessary? Doesn't seem to be needed for the other priors?

**Review by Balandat — COMMENTED** [review]
Thanks. Could you please also add some unit tests, similar to https://github.com/cornellius-gp/gpytorch/blob/master/test/priors/test_normal_prior.py ?

File: gpytorch/priors/__init__.py [hunk 4]
```diff
@@ -7,6 +7,7 @@
 from .torch_priors import (
     GammaPrior,
     HalfCauchyPrior,
+    HalfNormal,
```
Comment (Balandat) [inline gpytorch/priors/__init__.py:10]: This should be `HalfNormalPrior`. Hence the test failure

File: gpytorch/priors/torch_priors.py [hunk 5]
```diff
@@ -30,6 +30,25 @@ def expand(self, batch_shape):
         return NormalPrior(self.loc.expand(batch_shape), self.scale.expand(batch_shape))
 
 
+class HalfNormalPrior(Prior, HalfNormal):
+    """
+    Half-Normal prior.
+
```
Comment (Balandat) [inline gpytorch/priors/torch_priors.py:36]: get rid of the whitespace here

**Review by Balandat — COMMENTED** [review]
generally lg let's fix the test and flake8

**Review by Balandat — APPROVED** [review]
Thanks!
